### PR TITLE
clang-uml: init at 0.5.1

### PR DESCRIPTION
**Pull request by hatch01**
## Description of changes

Add the package clang-uml

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/cl/clang-uml/package.nix [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  pname = "clang-uml";
+  version = "0.5.1";
+
+  src = fetchFromGitHub {
+    owner = "bkryza";
+    repo = "clang-uml";
+    rev = finalAttrs.version;
+    hash = "sha256-AIWss4iIkht54WuVI1jTk3fC/w1UhelOsXMBucqWUlA=";
+  };
+
```
Comment (RossComputerGuy) [inline pkgs/by-name/cl/clang-uml/package.nix:21]: Please remove the duped space.

File: pkgs/by-name/cl/clang-uml/package.nix [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  cmakeFlags = [
+    "-DGIT_VERSION=${finalAttrs.version}"
+  ];
+
+  postInstall = ''
+    installShellCompletion --bash $src/packaging/autocomplete/clang-uml
+    installShellCompletion --zsh $src/packaging/autocomplete/_clang-uml
+  '';
+
```
Comment (RossComputerGuy) [inline pkgs/by-name/cl/clang-uml/package.nix:43]: Please remove the duped space.

File: pkgs/by-name/cl/clang-uml/package.nix [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+      The main idea behind the project is to easily maintain up-to-date diagrams within a code-base or document legacy code.
+      The configuration file or files for clang-uml define the types and contents of each generated diagram.
+      The diagrams can be generated in PlantUML, MermaidJS and JSON formats.
+    '';
+    maintainers = with maintainers; [ eymeric ];
+    homepage = "https://clang-uml.github.io/";
+    license = licenses.asl20;
+    platforms = platforms.all;
+  };
+
```
Comment (RossComputerGuy) [inline pkgs/by-name/cl/clang-uml/package.nix:58]: Please remove the duped space.

**Review by RossComputerGuy — COMMENTED** [review]
Looks great aside from a few duped empty lines.

**Review by RossComputerGuy — APPROVED** [review]

**Review by nim65s — APPROVED** [review]

File: pkgs/by-name/cl/clang-uml/package.nix [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    yaml-cpp
+  ];
+
+  cmakeFlags = [
+    "-DGIT_VERSION=${finalAttrs.version}"
+  ];
+
+  postInstall = ''
+    installShellCompletion --bash $src/packaging/autocomplete/clang-uml
+    installShellCompletion --zsh $src/packaging/autocomplete/_clang-uml
```
Comment (SuperSandro2000) [inline pkgs/by-name/cl/clang-uml/package.nix:40]:
```suggestion
    installShellCompletion --cmd clang-uml \
      --bash $src/packaging/autocomplete/clang-uml \
      --zsh $src/packaging/autocomplete/_clang-uml
```

**Comment by hatch01** [comment]
@pogobanane is it ok for you that I use your work on clang-uml for nix packaging in this pr here ([PR](https://github.com/bkryza/clang-uml/pull/273))
And should I use your wrapper, or is there a method to use the wrapper of clang directly in nixpkgs ?

**Comment by pogobanane** [comment]
> @pogobanane is it ok for you that I use your work on clang-uml for nix packaging in this pr here (https://github.com/bkryza/clang-uml/pull/273)
And should I use your wrapper, or is there a method to use the wrapper of clang directly in nixpkgs ?

We should merge this `clang-uml: init` first. Afterwards we can consider building a small, shared abstraction:

We already have that same wrapper here https://github.com/NixOS/nixpkgs/blob/0e2ddadf231ae164c333ac4d9e093c7ae21b970e/pkgs/development/tools/analysis/include-what-you-use/wrapper and here https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/tools/clang-tools/wrapper. However for clang-uml-wrapped, we run the binary in a different way, so we can't copy paste them. We could move the `buildcpath()` function and exports into a common file that is sourced by the app-respective wrappers. I'm not sure where to put such a `lib-clang-wrapper.sh` though. There is no precedent of putting it into `nixpkgs/lib`. We could perhaps integrate it into the llvmPackages, but i dont think that is worth the hassle.

**Review by pogoba — APPROVED** [review]
Looks good. I tested a previous version of this patch with my own code.

**Pull request merged**